otherlibs/stdune/fpath.mli

```diff
@@ -5,12 +5,20 @@ type mkdir_result =
   | Created  (** The directory was created. *)
   | Missing_parent_directory
       (** No parent directory, use [mkdir_p] if you want to create it too. *)
+  | Parent_not_directory  (** The parent exists but is not a directory. *)
 
 val mkdir : ?perms:int -> string -> mkdir_result
 
 type mkdir_p_result =
   | Already_exists  (** The directory already exists. No action was taken. *)
   | Created  (** The directory was created. *)
+  | Already_exists_not_directory
+      (** A file with the same name already exists but is not a directory. *)
```

Comment:
I don't think we need both constructors.

`Already_exists_not_directory of string` is enough. The argument specifies the file whether it's a path or not.

Reply:
You're right, much cleaner like this.